File: backend/database/models/booking.js
/*jshint esversion: 6 */
const mongoose = require('../mongoose');

const BookingSchema = new mongoose.Schema({
    _salonId: {
        type: mongoose.Types.ObjectId,
        require: [true, 'Salon id']
    },

    _customerId: {
        type: mongoose.Types.ObjectId,
        require: [true, 'Customer id']
    },

    _barberId: {
        type: mongoose.Types.ObjectId,
        require: [false, 'Barber id']
    },

    createdDate: { 
        type: Date,
        required: [false, 'Date']
    },

    bookingDate: { 
        type: Date,
        required: [false, 'Date']
    },

    bookingTime: { 
        type: String , 
        required: [false, 'Time']
    },
    
    info: { 
        type: String , 
        required: [false, 'Detail Booking info']
    },

    status: { 
        type: String , 
        required: [false, 'status of booking (completed/processing/canceled)']
    },
});

// get the current timestamp for the booking
BookingSchema.pre('save', async function(next) {

    this.createdDate = new Date().getTime();
    console.log('timestamp: ' + this.createdDate);
    next();
});

const Booking = mongoose.model('Booking', BookingSchema);
module.exports = Booking;

File: mobile/salonmobile/lib/models/Booking.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

class Booking {
  final String salonId;
  final String userId;
  final String barberId;
  final String createdDate;
  final String bookingDate;
  final String bookingTime;
  final String info;
  final String status;

  Booking(
      {
        @required this.salonId,
        @required this.userId,
        this.barberId,
        this.createdDate,
        this.bookingDate,
        this.bookingTime,
        this.info,
        this.status});

  factory Booking.fromJson(Map<String, dynamic> json) {

    return Booking(
        salonId: json['_salonId'],
        userId: json['_userId'],
        barberId: json['_barberId'],
        createdDate: json['createdDate'],
        bookingDate: json['bookingDate'],
        bookingTime: json['bookingTime'],
        info: json['info'],
        status: json['status']);
  }
}
